dblookupext/miniblocksHandler.go

```diff
@@ -280,3 +263,112 @@ func (mh *miniblocksHandler) removeMiniblockMetadata(
 
 	return mh.saveMiniblocksMetadata(miniblockHash, loadedMiniblockMetadata, header.GetEpoch())
 }
+
+func (mh *miniblocksHandler) loadAllExistingMiniblocksMetadata(miniblockHash []byte, epoch uint32) (*MiniblockMetadataV2, error) {
+	miniblockMetadataFirstEpoch, err := mh.loadExistingMiniblocksMetadata(miniblockHash, epoch)
+	if err != nil {
+		return nil, err
+	}
+	numTxHashes := 0
+	for _, mbInfo := range miniblockMetadataFirstEpoch.MiniblocksInfo {
+		numTxHashes += len(mbInfo.TxHashesWhenPartial)
+	}
+	if numTxHashes == 0 || numTxHashes == int(miniblockMetadataFirstEpoch.NumTxHashesInMiniblock) {
+		return miniblockMetadataFirstEpoch, nil
+	}
+
+	// we will try the next epoch, maybe we will find the rest of the miniblockMetadata info
+	miniblockMetadataNextEpoch, err := mh.loadExistingMiniblocksMetadata(miniblockHash, epoch+1)
+	if err != nil {
+		// no, return what we found
+		return miniblockMetadataFirstEpoch, nil
+	}
+	miniblockMetadataFirstEpoch.MiniblocksInfo = append(miniblockMetadataFirstEpoch.MiniblocksInfo, miniblockMetadataNextEpoch.MiniblocksInfo...)
+
+	return miniblockMetadataFirstEpoch, nil
+}
+
+func (mh *miniblocksHandler) getMiniblockMetadataByMiniblockHash(miniblockHash []byte) ([]*MiniblockMetadata, error) {
+	miniblockMetadata, err := mh.retrieveMiniblockMetadata(miniblockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return convertMiniblockMetadata(miniblockMetadata), nil
+}
+
+func (mh *miniblocksHandler) retrieveMiniblockMetadata(miniblockHash []byte) (*MiniblockMetadataV2, error) {
+	epoch, err := mh.epochIndex.getEpochByHash(miniblockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return mh.loadAllExistingMiniblocksMetadata(miniblockHash, epoch)
+}
+
+func (mh *miniblocksHandler) getMiniblockMetadataByTxHash(txHash []byte) (*MiniblockMetadata, error) {
+	miniblockHash, err := mh.miniblockHashByTxHashIndexStorer.Get(txHash)
+	if err != nil {
+		return nil, err
+	}
+
+	miniblockMetadata, err := mh.retrieveMiniblockMetadata(miniblockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	found := false
+	for _, mbInfo := range miniblockMetadata.MiniblocksInfo {
+		if findTxHashInMiniblockMetadataOnBlock(mbInfo, txHash) {
+			miniblockMetadata.MiniblocksInfo = []*MiniblockMetadataOnBlock{mbInfo}
```

Comment:
when found, could have placed these lines here:
```
mbData := convertMiniblockMetadata(miniblockMetadata)
	return mbData[0], nil
```
and get rid of the `found` variable

Reply:
agree. Simplified